package Protos;


message NeuronInfo {
	required string neuron = 1;
	optional string learning_rule = 2;
	required string act_func = 3;
	optional string tuning_curve = 4;
}

message LayerInfo {
	required uint32 size = 1;
	required uint32 id = 2;
	required NeuronInfo neuron = 3;
};


message Neuron {
	required uint32 id = 1;
	required double axon_delay = 2;
	required uint32 num_of_synapses = 3;
}

message Synapse {
	required double w = 1;
	required uint32 id_pre = 2;
	required double dendrite_delay = 3;
	required string const_name = 4;
}



